### DISCUSSION TITLE
Clippy

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#168] [Q&A] [answered]
I'd like nci to execute `cargo clippy` instead of `cargo build`. Not sure there's an option for that...

**Comment by 90-008** [comment]
why would you want to replace the build with clippy? either way i don't think there is an easy way to do this right now, the commands are hardcoded in dream2nix and there isn't really an option to change the commands out, only the flags. you can try overriding the `checkPhase` and set it to something like `cargo clippy --profile whatever --package my-crate-name` and it should maybe work?

**Reply by mightyiam** [reply]
Hrm... wait, no. `cargo clippy` is a superset of `cargo check`, not `cargo build`. I was confused about that. Sorry.

Well, in any case, clippy as a substitute for `check` is super popular, and for good reason, right?

So, perhaps an option could be provided by nci for that?

Perhaps the following, to keep what seems to be a high level of abstraction nci is aiming for?

`perSystem.nci.projects.<name>.clippy.enable`
`perSystem.nci.crates.<name>.clippy.enable`

**Reply by 90-008** [reply]
i'll probably end up just adding an option to change the cargo command being used, that seems like the best option here to me

**Reply by mightyiam** [reply]
Does that design compose well with other options you're considering?

**Reply by 90-008** [reply, marked as answer]
i looked into it a bit more and just added a useClippy option. the changing cargo command i added to dream2nix, so it will also work in nci fine, but since clippy is the only check thing i have ever seen anyone use other than just cargo check, i decided to add useClippy as an option

**Reply by mightyiam** [reply]
Brilliant, thank you!
